### PR TITLE
Added readme

### PR DESCRIPTION
**Pull request by andyferris**
@OmriRoames @c42f

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,97 @@
...
+problems painless and simple. It provide these useful features:
+
+1. A set of linear equation solvers, to perform linear least-squares optimization steps.
+2. A set of *M*-estimators for robust estimation. An estimator is used to remove
+   outliers by reducing the cost (loss) function for abnormally large residuals, and
+   gets its name from the statistical method of estimating parameters from
+   certain statistical distributions (e.g. Gaussian statistics correspond to the
+   standard *L2*-estimator, etc).
+3. A high-level optimization routine that performs iteratively reweighted least-squares,
+   solving the effectively non-linear optimization of the cost/loss function by
```
Comment (c42f) [inline README.md]: ~~effectively~~

File: README.md [hunk 2]
```diff
@@ -1,3 +1,97 @@
...
+   a series of quadratic minimization steps.
+
+### Solvers
+
+The package has three in-built solvers - *QR* factorization of the ordinary
+least-squares problem, the normal-form of the least-squares problem, and conjugate
+gradient on the former.
+
+We are trying to find *x* such that *|r|* is minimized where the residual *r = Ax - b* (i.e. find the *x* so
```
Comment (c42f) [inline README.md]: I'd probably use `norm(r)` here, or `||r||`.  The single bars just look odd to me.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,97 @@
...
+that *Ax* is as close as possible to *b* - if *A* is invertible, the solution is
+*inv(A).b* and *r = 0*). The equivalent "normal" form is *A'Ax - A'b*, where *A'A* is now
```
Comment (c42f) [inline README.md]: `inv(A).b` -> `A\b` ?  I guess I'm erring on the side of julia notation rather than math here.

Comment (c42f) [inline README.md]: About the normal form, I'd perhaps say something like

"The solution may be found directly using the "normal equations" _A'A x = A'b_ where _A'A_ is now a square, Hermitian matrix.  This is more poorly numerically conditioned, but can be faster depending on the sparsity of _A_.

(Is my comment about sparsity correct?  I'm not sure whether a non-sparse A'A implies the QR decomposition of A will also be bad, or whether QR can give a big win in that case...)

File: README.md [hunk 4]
```diff
@@ -1,3 +1,97 @@
...
+a square, Hermitian matrix, and depending on the dimensions of *A* and the
+conditioning of the systems, this may be easier or harder to solve.
+
+A *weighted* least-squares problem introduces a weight vector *w* which is
+multiplies *r* element-wise to scale some residual elements to be larger or smaller,
+increasing or decreasing their importance in the optimization procedure (which
+is what lets us reduce the impact of outliers on the solution).
```
Comment (c42f) [inline README.md]: Might be worth noting that the equations are ||W(Ax - b)|| where `W = diagm(weights)` is a diagonal weight matrix, or some such thing.  Also possibly worth noting that weights is simply the inverse standard deviations in the normally distributed case.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,97 @@
...
+
+1. `L2Estimator()`, *cost* = `sum(r.^2)`
+2. `L1Estimator()`, *cost* = `sum(abs(r))`
+3. `L1L2Estimator()`, behaves *L2* for *r << 1*, and *L1* for *r >> 1*
+4. `FairEstimator(c)`, behaves *L2* for *r << c*, and *L1* for *r >> c*
+5. `HuberEstimator(c)`, behaves *L2* for *r << c*, and *L1* for *r >> c*
+6. `CauchyEstimator(c)`, behaves *L2* for *r << c*, and logarithmic for *r >> c* (non-convex)
+7. `GemanEstimator()`,  behaves *L2* for *r << 1*, and Lorentzian for *r >> 1* (non-convex)
+8. `WelschEstimator(c)`, behaves *L2* for *r << c*, and Gaussian for *r >> c* (non-convex)
+9. `TukeyEstimator(c)`, behaves *L2* for *r < c*, and complete cutoff for *r > c* (non-convex)
```
Comment (c42f) [inline README.md:83]: Nice summary with the asymptotics listed.  Github markdown rendering seems broken for some reason.

**Comment by c42f** [comment]
Looks pretty good.  A nice-to-have might be a quickstart example at the top showing off the main features, eg applying a MultiEstimator to a regularized least squares problem with outliers.

File: README.md (hunk 3 above)
Comment (andyferris) [inline README.md]: About `A\b`, that's good Julia even if `A` is not invertible, and is a kind-of magical function that does different thing depending on `A` and `b`. Here, I was trying to specifically give people a handle of what happens in a very simple case in mathematical notation.

**Pull request merged**